### PR TITLE
Migrate to gson

### PR DESCRIPTION
**Pull request by lettucegoblin**
Migrated all necessary places where JSONObject still exists. Still needs cleanup. Deleted httphandlertest since we're migrating.

Also setup groundwork for protobuffers: 
- test file and proper config.
- pom.xml, added version variables and linux profile for github cli

### CODE REVIEW

File: src/main/java/org/fibsters/BufferedImageTypeAdapter.java [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        }
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        String imageExtension = "png";
+        if (this.imageType == ImageType.PNG) {
+            imageExtension = "png";
+        }
+        ImageIO.write(image, imageExtension, bos);
+        byte[] imageBytes = bos.toByteArray();
+        String imageBase64 = Base64.getEncoder().encodeToString(imageBytes);
+        out.value(imageBase64);
```
Comment (iron768) [inline src/main/java/org/fibsters/BufferedImageTypeAdapter.java:41]: Using GSON makes this a lot less headache inducing, nice.

Comment (lettucegoblin) [reply src/main/java/org/fibsters/BufferedImageTypeAdapter.java:41]: Right? I felt insane before.

File: src/main/java/org/fibsters/InputPayloadImpl.java [hunk 2]
```diff
@@ -32,21 +42,78 @@ public class InputPayloadImpl implements InputPayload {
...
+
+                break;
+            default:
+                throw new Exception("Error: Must be valid directive type" + getListOfDirectiveTypesAsStringLol());
+        }
+    }
+
+    private String getListOfDirectiveTypesAsStringLol() {
+        List<DirectiveType> directiveTypes = Arrays.asList(InputPayloadImpl.DirectiveType.values());
+        List<String> directiveTypeNames = directiveTypes.stream()
```
Comment (iron768) [inline src/main/java/org/fibsters/InputPayloadImpl.java:87]: Nice use of Java's lambda expressions

File: src/main/java/org/fibsters/OutputPayloadImpl.java [hunk 3]
```diff
@@ -2,24 +2,27 @@
...
+import org.fibsters.util.UniqueIDGenerator;
 
 import java.awt.image.BufferedImage;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
 public class OutputPayloadImpl implements OutputPayload {
-
+    private String uniqueID;
```
Comment (iron768) [inline src/main/java/org/fibsters/OutputPayloadImpl.java:13]: Why store as a String object and not a UUID object?

**Review by iron768 — APPROVED** [review]
Lgtm

File: src/main/java/org/fibsters/BufferedImageTypeAdapter.java (hunk 1 above)
Comment (ndg8743) [reply src/main/java/org/fibsters/BufferedImageTypeAdapter.java:41]: we're all a lil insane right???

File: src/main/java/org/fibsters/OutputPayloadImpl.java (hunk 3 above)
Comment (ndg8743) [reply src/main/java/org/fibsters/OutputPayloadImpl.java:13]: possible maybe oopsie dasiy

Comment (lettucegoblin) [reply src/main/java/org/fibsters/OutputPayloadImpl.java:13]: That would be better for decoupling we'll do that at some point 100%

File: src/main/java/org/fibsters/InputPayloadImpl.java (hunk 2 above)
Comment (lettucegoblin) [reply src/main/java/org/fibsters/InputPayloadImpl.java:87]: I hope you also like the function name

**Pull request merged**